docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,540 @@
+   
+   \frac{\partial h_k}{\partial t} + \nabla \cdot (h_k \mathbf{u}_k) = 0
+
+   \frac{\partial (h_k \phi_k)}{\partial t} + \nabla \cdot (h_k \mathbf{u}_k \phi_k) = [D_h^{\phi}]_k + [D_{v}^{\phi}]_k + F_k^{\phi}
+   
+The time-stepping algorithm (RK4 or split-explicit) yields the updated 
+variables :math:`u_k^{lg},h_k^{lg},\phi_k^{lg}`, where the superscript
+*lg* is used to designate the values after the Lagrangian step.
+
+Note that the vertical mixing terms :math:`D_h^u, D_v^h, D_h^{\phi}, D_v^{\phi}` 
+are retained here. We opt to compute these terms prior to remapping as this 
+allow for future development in which the dynamics are subcycled relative to 
+the thermodynamics and remapping is scheduled on the thermodynamic timestep. 
+This computation of vertical mixing terms prior to remapping is similar to 
+both MOM6 and HYCOM. We anticipate that there could be a trade-off between (a)
+loss of accuracy of vertical mixing terms when their computation precedes 
+remapping due to grid deformation and (b) loss of accuracy when their 
+computation follows remapping due to remapping errors in vertical gradients of 
+prognostic variables. We do not intend to test this at this time.
+
+The target grid needs to be determined after the solution for prognostic 
+variables so that the vertical Lagrangian-remapping method is general enough to
+be used with coordinate systems that depend on the ocean state (this includes 
+the z-star coordinate system in which SSH perturbations are vertically 
+distributed between layers). We do not present an algorithmic design for 
+regridding to coordinate systems not already supported in MPAS-Ocean, as this 
+will be the subject of future development. For now, the target grid based on a 
+constant set of z-levels that are specified at initialization.
+
+For the regridding step, layer thicknesses are set according to the target 
+grid, conserving volume:
+
+.. math::
+
+   h_k^{t+1} = h_k^{target}
+   
+   \sum_{k=1}^{kmax}h_k^{t+1} = \sum_{k=1}^{kmax}h_k^{lg}
+
+
+For the remapping step, velocities (edge-normal) and scalars are remapped to 
+the target grid, conserving volume flux and scalar concentration:
+
+.. math::
+
+   \sum_{k=1}^{kmax} u_k^{t+1} h_k^{t+1} = \sum_{k=1}^{kmax} u_k^{lg} h_k^{lg}
+   
+   \sum_{k=1}^{kmax} \phi_k^{t+1} h_k^{t+1} = \sum_{k=1}^{kmax} \phi_k^{lg} h_k^{lg}
+
+The vertical velocity across layer interfaces may be computed anytime after 
+regridding. It can be computed as 
+
+.. math::
+
+   w = - \nabla \cdot (h_k \mathbf{u}_k) - (h_k^{t+1} - h_k^t)/dt
+
+or
+
+.. math::
+
+   w = (h_k^{t+1} - h_k^lg)/dt
+
+The choice between the two is discussed in the Implementation section.
+
+Algorithm Design: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+*Discuss damping (in time) function here*
```

Comment:
@dengwirda This might not be the right place to discuss the damping function for the rate of grid movement, but I wasn't sure it should be a separate requirement either.

Reply:
In cases where we're remapping back to known standard grids (i.e. z*, etc) there's no need for any smoothing of the target grid in time or space, since z^{t+1} / h^{t+1} can be determined diagnostically from SSH, etc.

In the "full" hybrid coordinate version, where the target grid is determined via heuristics, the spatial and time-based smoothing will come into play. Perhaps these details are best left for future revisions at this stage though?

Reply:
Good point. Maybe this belongs with the next "hybrid-grid" design doc.

Reply:
Now addressed by https://github.com/MPAS-Dev/MPAS-Model/pull/763/commits/6c1f1a91d466c47f049d3c4db0cc46ad7f9d585d which removes discussion of smoothing from this design doc.